posix.mak

```diff
@@ -551,30 +538,28 @@ d.tag : chmgen.d $(STABLE_DMD) $(ALL_FILES) phobos-release druntime-release
 #
 # - This transforms assert(a = b) to writeln(a); // b
 # - It creates a copy of Phobos to apply the transformations
+# - All "d" files are piped through the transformator,
+#   other needed files (e.g. posix.mak) get copied over
 ################################################################################
 
-$(GENERATED):
-	mkdir -p $@
+ASSERT_WRITELN_BIN = $(GENERATED)/assert_writeln_magic
 
-# --update allows to copy only the newer files and thus only propagate these
-#  changes
-HAS_RSYNC := $(shell command -v rsync 2> /dev/null)
+$(ASSERT_WRITELN_BIN): assert_writeln_magic.d $(DUB)
+	@mkdir -p $(dir $@)
+	$(DUB) build --single $<
```

Comment:
Note that using `DUB` without a specified `DMD` executable apparently broke Martin's build script. 
This is the current "status quo" and I am waiting for his feedback on how to resolve this at https://github.com/dlang/dlang.org/pull/1592
(so this shouldn't block this PR and #1592 can easily be rebased)